pkg/controller/unifiedpushserver/unifiedpushserver.go

```diff
@@ -91,6 +91,18 @@ func newUnifiedPushServerDeployment(cr *pushv1alpha1.UnifiedPushServer) (*appsv1
 				},
 				Spec: corev1.PodSpec{
 					ServiceAccountName: cr.Name,
+					InitContainers: []corev1.Container{
+						{
+							Name:            POSTGRES_CONTAINER_NAME,
+							Image:           postgresql.image(),
+							ImagePullPolicy: corev1.PullAlways,
+							Command: []string{
+								"/bin/sh",
+								"-c",
+								"until nslookup example-unifiedpushserver-postgresql; do echo waiting for service; sleep 2; done;",
```

Comment:
Unfortunately `nslookup` won't be enough here, since that should return successfully once the `Service` is created. What we care about here though is that there's anything ready for requests behind that `Service` (i.e. Pods that have been marked as Ready that match the selector in the service).

I've just tested that it's not enough also, by scaling down the Postgresql to 0 pods, and then recreating the UPS pod: it passes the initialization phase straight away, even though what we want is for it to fail.

Reply:
:+1: thanks for testing